src/bio/bio_buffer.c

```diff
@@ -1105,6 +1112,16 @@ nvme_rw(struct bio_desc *biod, struct bio_rsrvd_region *rg)
 	if (daos_io_bypass & IOBP_NVME)
 		return;
 
+	/* No locking for BS state query here is tolerable */
+	if (bxb->bxb_blobstore->bb_state == BIO_BS_STATE_FAULTY) {
+		D_ERROR("Blobstore is marked as FAULTY.\n");
```

Comment:
[minor] this might generate a lot of error messages if it is faulty, maybe change it to debug or just output once.